### PR TITLE
silence #file to #filePath warnings

### PR DESCRIPTION
**Pull request by weissi**
Motivation:
    
    Swift 5.3 will keep `#file` == `#filePath` but starts warning if you
    pass `#file` to an argument that's `#filePath` defaulted (XCTest).
    
    Modification:
    
    - Replace `#file` with `(#file)` to silence the warning. In Swift 5.3
      `#file` == `#filePath`.
    
    Result:
    
    No warnings

### CODE REVIEW

File: Tests/LoggingTests/TestLogger.swift [hunk 1]
```diff
@@ -292,3 +300,13 @@ internal struct TestLibrary {
         }
     }
 }
+
+#if compiler(>=5.3)
+internal func fullFilePath(_ filePath: StaticString = #filePath) -> StaticString {
```
Comment (weissi) [inline Tests/LoggingTests/TestLogger.swift:305]: this is here because `XCTest` wants a `StaticString` and not a `String`.

**Comment by ktoso** [comment]
Formatting issues: https://ci.swiftserver.group/job/swift-log-swift50-prb/162/console (whitespace)

PR LGTM, ok to keep compatibility and no surprising changes I guess.
In the long run honestly path is what we'll want but we need to discuss and _get_ #module etc... Will make a separate ticket about it though.

**Comment by weissi** [comment]
I'm deliberately not fixing the formatting right now because I think this is blocked on https://bugs.swift.org/browse/SR-12934

**Comment by weissi** [comment]
@ktoso PR totally different now. Swift 5.3 actually keeps the behaviour, we just need to silence the  XCTest warnings for now.

File: .swiftformat [hunk 2]
```diff
@@ -11,3 +11,5 @@
 --ifdef no-indent
 
 # rules
+
+--disable redundantParens # https://github.com/nicklockwood/SwiftFormat/issues/638
```
Comment (ktoso) [inline .swiftformat:15]: 👍

File: Tests/LoggingTests/TestLogger.swift [hunk 3]
```diff
@@ -292,3 +300,14 @@ internal struct TestLibrary {
...
```
Comment (ktoso) [inline Tests/LoggingTests/TestLogger.swift:305]: Why do we need this one in tests?
If #file == #filePath but just the warning around `= #file` sounds like there should be no difference on how we treat this in tests and the lib itself?

File: Sources/Logging/Logging.swift [hunk 4]
```diff
@@ -235,7 +235,7 @@ extension Logger {
     ///    - message: The message to be logged. `message` can be used with any string interpolation literal.
     ///    - metadata: One-off metadata to attach to this log message
     ///    - file: The file this log message originates from (there's usually no need to pass it explicitly as it
-    ///            defaults to `#file`).
+    ///            defaults to `#file` (until Swift 5.2) / `#filePath` (from Swift 5.3)).
```
Comment (ktoso) [inline Sources/Logging/Logging.swift:238]: Comment is semi true I guess... Would it be better to remove it? It's not like _we_ make that `if ...` but it's just Swift's behavior, so why should _we_ document that here if observed behavior is the same?

Comment (weissi) [reply Sources/Logging/Logging.swift]: good catch, thank you!

**Comment by ktoso** [comment]
Status update — I think this likely will be fine but I’ve ran out of time in the day to get it to completion in my project to really see it work out end to end.

It makes the label pretty useless but I guess we’ll agree that’s what it is. I’ll post a review tomorrow

**Comment by weissi** [comment]
> Status update — I think this likely will be fine but I’ve ran out of time in the day to get it to completion in my project to really see it work out end to end.
> 
> It makes the label pretty useless but I guess we’ll agree that’s what it is. I’ll post a review tomorrow

I thought about `label` more and I actually now think it's useful. It's basically orthogonal to `source`.

Let's consider a whole system and let's assume the system offers 2 services, let's call them service A and service B. Services A & B both use a multitude of sub-subsystems. Then everything that belongs to Service A gets `label=MyApp.ServiceA` and B gets `label=MyApp.ServiceB`. If now both A & B use say AsyncHTTPClient, then the label suddenly becomes useful.

- `source=AHC` gives us AHC calls from both A & B which might be annoying if I'm a developer working only on A
- `source=AHC & label=MyApp.ServiceA` however gives me exactly what I want.

**Comment by weissi** [comment]
@ktoso I think we're discussing this on the wrong issue actually :P

**Comment by ktoso** [comment]
> @ktoso I think we're discussing this on the wrong issue actually :P

Whoops, you're right 😱 

Moving to https://github.com/apple/swift-log/pull/135

**Pull request merged**